### ISSUE TITLE
Libs/Strings: Errors parsing Japanese text

### ISSUE THREAD

**Issue by mazdarxseven** [Windower/Lua#2396]
Original pull request here for context:
https://github.com/Windower/Lua/pull/2394

Doing some simple tests with //send to my alt for some SCH spells,
Works:
`send Alt /ja "女神降臨の章" <me>`
(Accession)

Doesn't work, throwing an error on strings.lua line 91: attempt to perform arithmetic on local 'length' (a boolean value), as described in the pull request:
`send Alt /ma "鼓舞激励の策" <me>`
(Adloquium)

The version published about 19 hours ago https://github.com/Windower/Lua/pull/2395/commits/7d7d644133cbfe456f2c99948fa3f09f89246182 fixed adloquium, but when I tried:
`send Alt /ja "机上演習" <me>`
(Sublimation)

I got an error in strings.lua line 88: Invalid code point

**Comment by Nifim** [comment]
Another report with some more details from a discord post [#support - Vires](https://discord.com/channels/338590234235371531/338713865812574219/1305642420788006943):
> Some JP friends are having some issues with the latest windower update.  It seems to be related to how JP characters are 2 byte characters, versus US alphabet of 1 byte characters.  I'll do my best to explain the issue for them.
>
> First they use an addon called Magicreplace, which replaces JP characters with english translations to send to other addons that dont support JP inputs.  Magicreplace creator retired but it seems they were able to find a fix for their error (magicreplace　で attempt to perform arithmetic on a nil value).  This was fixed by changing:
This: command:find(magicname,1,1) to this: command:find(magicname,1,true)
>
> But that creates a new error in Strings.lua, error on line 91, which brings me here.
>
> This was also fixed by giving strings a 2 byte interpretation by changing
This on line 114: byte == 0xFD and 6 to this: byte == 0xFD and 6 or 2 though it still can not handle old or uncommon characters like 「解」and still gives an error in Strings.lua on line 91.
tl;dr seems to be strings.lua is having trouble with JP characters after the new update.

**Comment by z16** [comment]
We pushed an update that is meant to resolve this. Can you test it again?

**Comment by mazdarxseven** [comment]
I tested out the new version today and from what I can tell, the issue appears to be resolved. Adloquium and sublimation worked without a hitch along with any other commands I threw at my alt with send.
Thanks for the tip about Magicreplace as well. I'd just about given up on it since the author is retired, but looks like I'll be able to continue using it. Cheers!

**Comment by z16** [comment]
Great, thanks for confirming!
